//
//  replicate.proto
//  replicatedlog
//
//  Created by Liudmyla Patenko on 25.10.2022.
//

syntax = "proto3";

package replicatedlog;

service ReplicateService {
    rpc appendMessage(MessageItem) returns(ReplicateResponce) {}
    rpc getLastMessageId(ReplicateParamStub) returns(LastMessageId) {}
}

message ReplicateResponce {
    int32 res = 1;
}

message MessageItem {
    int32 id = 1;
    string text = 2;
}

message LastMessageId {
    int32 id = 1;
}

message ReplicateParamStub {
    int32 id = 1;
}
